internal/declarative/v2/reconciler.go

```diff
@@ -406,6 +406,10 @@ func (r *Reconciler) checkTargetReadiness(
 		return ErrInstallationConditionRequiresUpdate
 	}
 
+	if !manifest.GetDeletionTimestamp().IsZero() {
```

Comment:
this condition is related to manifest CR, it's not related to `checkTargetReadiness`, should not inside this function, and also consider renaming `checkTargetReadiness` to something more intuitive, like `checkDeploymentState`

Reply:
I will refactor the function to split the deployment check and the manifest status setting into two different functions

Reply:
check the comment here https://github.com/kyma-project/lifecycle-manager/pull/1628/files#r1658759549